pyproject.toml

```diff
@@ -1,30 +1,88 @@
-  "if 0:",
-  "if __name__ == .__main__.:", # Don't complain about mypy-specific code
-  "if TYPE_CHECKING:",
+[tool.poetry]
+name = "sceptre"
+version = "4.0.2"
+packages = [{ include = "sceptre" }]
+readme = "README.md"
+homepage = "https://github.com/Sceptre/sceptre"
+repository = "https://github.com/Sceptre/sceptre"
+documentation = "https://docs.sceptre-project.org"
+authors = ["Sceptre <sceptreorg@gmail.com>"]
+description = "An AWS Cloud Provisioning Tool"
+keywords = ["aws", "cloud", "devops", "infrastructure", "tools", "cli"]
+license = "Apache-2.0"
+classifiers = [
+  "Development Status :: 5 - Production/Stable",
+  "Intended Audience :: Developers",
+  "Natural Language :: English",
+  "Environment :: Console",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11"
 ]
-ignore_errors = true
-fail_under = 90.0
-show_missing = true
-
-[tool.pytest.ini_options]
-# The "--cov" option will disable the pycharm debugger.  To enable the debugger either
-# remove this flag or add the "--no-cov" to your pycharm's pytest run configuration.
-# https://github.com/pytest-dev/pytest-cov/issues/131
-# https://stackoverflow.com/questions/34870962/how-to-debug-py-test-in-pycharm-when-coverage-is-enabled
-addopts = "-s --cov --cov-report term --cov-report term-missing --cov-report xml"
+
+[tool.poetry.scripts]
+"sceptre" = "sceptre.cli:cli"
+
+[tool.poetry.plugins."sceptre.hooks"]
+"asg_scheduled_actions" = "sceptre.hooks.asg_scaling_processes:ASGScalingProcesses"
+"cmd" = "sceptre.hooks.cmd:Cmd"
+
+[tool.poetry.plugins."sceptre.resolvers"]
+"environment_variable" = "sceptre.resolvers.environment_variable:EnvironmentVariable"
+"file_contents" = "sceptre.resolvers.file_contents:FileContents"
+"stack_output" = "sceptre.resolvers.stack_output:StackOutput"
+"stack_output_external" = "sceptre.resolvers.stack_output:StackOutputExternal"
+"no_value" = "sceptre.resolvers.no_value:NoValue"
+"select" = "sceptre.resolvers.select:Select"
+"stack_attr" = "sceptre.resolvers.stack_attr:StackAttr"
+"sub" = "sceptre.resolvers.sub:Sub"
+"split" = "sceptre.resolvers.split:Split"
+"join" = "sceptre.resolvers.join:Join"
+
+[tool.poetry.plugins."sceptre.template_handlers"]
+"file" = "sceptre.template_handlers.file:File"
+"s3" = "sceptre.template_handlers.s3:S3"
+"http" = "sceptre.template_handlers.http:Http"
+
+[tool.poetry.dependencies]
+python = ">=3.8,<3.12"
+boto3 = ">=1.3.0,<2"
+click = ">=7.0,<9.0"
+cfn-flip = ">=1.2.3,<2.0"
+deepdiff = ">=5.5.0,<6.0"
+deprecation = ">=2.0.0,<3.0"
+jinja2 = ">=3.0,<4"
+jsonschema = ">=3.2,<3.3"
+networkx = ">=2.6,<2.7"
+packaging = ">=16.8,<22.0"
+pyyaml = ">=5.1,<6.0"
+sceptre-cmd-resolver = ">=1.1.3,<3"
+sceptre-file-resolver = ">=1.0.4,<2"
+colorama = ">=0.2.5,<0.4.4"
+troposphere = { version = ">4,<5", optional = true }
+
+[tool.poetry.extras]
+runtime = ["troposphere"]
```

Comment:
So what's going on here? How does this combine with the `troposphere` above?

Reply:
done

Reply:
Why is this extra called "runtime"? This means to install this extra explicitly you need to invoke `poetry install -E runtime` which doesn't make sense. I think it should be called `troposphere`.

Reply:
done